primitives/src/portfolio.rs

```diff
@@ -0,0 +1,46 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymeshAssociation/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+use crate::{Balance, NFTs, Ticker};
+
+/// Describes what should be moved between portfolios. It can be either fungible or non-fungible tokens.
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo)]
+pub struct Fund {
```

Comment:
Moving the type definition to primitives was needed to avoid adding the portfolio pallet as a dependency in the common crate. This also means that Memo would have to be moved as well, but to avoid that I created a new Memo type as a placeholder. As an alternative we can also move things from common to the portfolio pallet as it's done in the settlement crate.

Reply:
We have two other `Memo` types.  `polymesh_common_utilities::balances::Memo` and `pallet_settlement::InstructionMemo` that wrap the same 32 bytes.

Can we move/re-use the memo type from `pallets/common/src/traits/balances.rs`?